File: java/src/main/java/cn/wpush/sdk/HttpTransport.java
package cn.wpush.sdk;

import java.io.IOException;
import java.util.Map;

/** Injectable HTTP transport for testability. */
public interface HttpTransport {
  /** POST JSON and return status + body text. */
  Response post(String url, Map<String, String> headers, String jsonBody) throws IOException;

  final class Response {
    public final int status;
    public final String body;

    public Response(int status, String body) {
      this.status = status;
      this.body = body;
    }
  }
}

File: php/src/HttpTransport.php
<?php
declare(strict_types=1);

namespace WPush;

interface HttpTransport
{
    /**
     * @param array<string,string> $headers
     * @return array{status:int,body:string}
     */
    public function post(string $url, array $headers, string $jsonBody): array;
}
